src/client/dbif.cc

```diff
@@ -565,25 +646,41 @@ dbif::InfoPromise* NCWrapper::handleBlobDataRequest(
 }
 
 dbif::InfoPromise* NCWrapper::handleChunkDataRequest(data::NodeID id, bool sub) {
-  uint64_t qid = nc_->nextQid();
+  uint64_t qid_data = nc_->nextQid();
+  uint64_t qid_children = nc_->nextQid();
```

Comment:
Hmm, can we make `nextQid` use some atomic counter inside to avoid risk of duplicate qid if we have situation where two threads use it concurrently?